lms/djangoapps/discussion/static/discussion/js/discussion_board_factory.js

```diff
@@ -38,68 +35,46 @@
                 discussion = new window.Discussion(threads, {pages: threadPages, sort: sortPreference});
                 courseSettings = new window.DiscussionCourseSettings(options.course_settings);
 
-                // Create the new post view
-                newPostView = new NewPostView({
-                    el: $('.new-post-article'),
-                    collection: discussion,
-                    course_settings: courseSettings,
-                    mode: 'tab'
+                discussionBoardView = new DiscussionBoardView({
+                    el: $('.discussion-board'),
+                    discussion: discussion,
+                    courseSettings: courseSettings
                 });
-                newPostView.render();
 
                 // Set up the router to manage the page's history
                 router = new DiscussionRouter({
                     courseId: options.courseId,
                     discussion: discussion,
                     courseSettings: courseSettings,
-                    newPostView: newPostView
+                    discussionBoardView: discussionBoardView
                 });
-                router.start();
 
-                // Initialize and render search box
-                searchBox = new DiscussionSearchView({
-                    el: $('.forum-search'),
-                    threadListView: router.nav
-                }).render();
-
-                // Initialize and render breadcrumbs
-                BreadcrumbsModel = Backbone.Model.extend({
-                    defaults: {
-                        contents: []
-                    }
+                // Create the new post view
+                newPostView = new NewPostView({
+                    el: $('.new-post-article'),
+                    collection: discussion,
+                    course_settings: courseSettings,
+                    mode: 'tab'
                 });
+                newPostView.render();
+                router.newPostView = newPostView;
```

Comment:
@andy-armstrong This was my question yesterday. The new post view has to stick with router. Did I miss anything?